[adapters] Shorten CI

PatKamin opened the pull request.

lukaszstolarczuk commented:
I was just about to do some shortening myself 😉 Can you perhaps include my commit here: https://github.com/lukaszstolarczuk/unified-runtime/commit/823d41d4cab279a66404399e84de0dae2870f922

This is what we discussed yesterday.

PatKamin commented:
> I was just about to do some shortening myself 😉 Can you perhaps include my commit here: [lukaszstolarczuk@823d41d](https://github.com/lukaszstolarczuk/unified-runtime/commit/823d41d4cab279a66404399e84de0dae2870f922)
> 
> This is what we discussed yesterday.

Done. Modified commit msg a bit, moved the info about the target branch to the PR title.

lukaszstolarczuk reviewed (approved):
thx!

lukaszstolarczuk commented on .github/workflows/coverity.yml:
```diff
@@ -31,20 +31,12 @@ jobs:
         run: pip install -r third_party/requirements.txt
 
       - name: Configure CMake
         run: cmake -B $WORKDIR/build -DUR_ENABLE_TRACING=ON -DUR_DEVELOPER_MODE=ON -DUR_BUILD_TESTS=ON -DUR_FORMAT_CPP_STYLE=ON -DUMF_ENABLE_POOL_TRACKING=ON
```
`UR_FORMAT_CPP_STYLE=OFF`

PatKamin replied:
Done

lukaszstolarczuk reviewed (commented):
are you planning on doing similar PR on main branch...? 😉

pbalcer commented:
This is still in draft, is that intentional?
